### PR TITLE
Support multiple accounts in a system

### PR DESCRIPTION
**Pull request by justqyx**
Hi @reverbdotcom

I have implemented a `ChargeGateway` to support multiple accounts in a system, this is useful when our system has multiple Affirm accounts.

So we should instantiate a Affirm::ChargeGateway object before creating/retrieving a charge.

```ruby
charge_gateway = Affirm::ChargeGateway.new(
  environment: 'live', # or sandbox
  public_key: 'xxxx',
  secret_key: 'xxxx'
)
```

## Creating/authorizing a charge

```ruby
charge = charge_gateway.create('checkout_token')
```

## Retrieving a charge

```ruby
charge = charge_gateway.retrieve("ABCD-ABCD")
```

### CODE REVIEW

**Comment by mcostanza** [comment]
Thanks for the pull request!

Would you mind providing context on why multiple Affirm accounts might be necessary? I think it may be a more common use case to have one account per application. 

In order to minimize changes to the gem's public API, I suggest refactoring with the ability to configure a default gateway, and providing the ability to inject a non-default gateway when creating a charge:

```
Affirm::API.default_charge_gateway = Affirm::ChargeGateway.new(
  environment: 'live',
  public_key: 'xxx',
  secret_key: 'xxx'
)

charge_1 = Affirm::Charge.create("AAAA-AAAA") # uses the default gateway

other_gateway = Affirm::ChargeGateway.new(
 environment: 'live',
 public_key: 'other_key',
 secret_key: 'other_secret'
)

charge_2 = Affirm::Charge.create("BBBB-BBBB", gateway: other_gateway)
```

File: README.md [hunk 1]
```diff
@@ -1,11 +1,13 @@
 # affirm-ruby
+
 Ruby client library for integrating with Affirm financing (https://www.affirm.com/)
 
 Requires Ruby 2.1 or greater.
 
-[![Build Status](https://travis-ci.org/reverbdotcom/affirm-ruby.png?branch=master)](https://travis-ci.org/reverbdotcom/affirm-ruby)
+[![Build Status](https://travis-ci.org/justqyx/affirm-ruby.png?branch=master)](https://travis-ci.org/justqyx/affirm-ruby)
```
Comment (mcostanza) [inline README.md]: Please switch this back to the `reverbdotcom` repo.

File: lib/affirm/charge_event.rb [hunk 2]
```diff
@@ -1,20 +1,20 @@
 module Affirm
   class ChargeEvent
     attr_reader :id, :transaction_id, :type, :created,
-      :amount, :fee, :order_id, :shipping_carrier, :shipping_confirmation,
-      :fee_refunded
+                :amount, :fee, :order_id, :shipping_carrier, :shipping_confirmation,
```
Comment (mcostanza) [inline lib/affirm/charge_event.rb]: Could this and other spacing/formatting changes be removed? They're not relevant to the PR.

File: README.md (hunk 1 above)
Comment (justqyx) [reply README.md]: OK, I'll make change later.

File: lib/affirm/charge_event.rb (hunk 2 above)
Comment (justqyx) [reply lib/affirm/charge_event.rb]: OK, I'll make change later.

**Comment by justqyx** [comment]
@mcostanza 

As you said, in a common application, it's not necessary to support multiple affirm accounts. However, it might need to build a common platform for reducing develop resource cost in a company which usually has many projects and some of them need support payment.

The project I'm developing recently needs to apply the payment ability for other projects of my company. 

The [braintree ruby gem](https://github.com/braintree/braintree_ruby) support this function in it's internal which is not declared in README. And this design inspires me to build a `ChargeGateway` for this projects.

```ruby
require "rubygems"
require "braintree"

Braintree::Configuration.environment = :sandbox
Braintree::Configuration.merchant_id = "your_merchant_id"
Braintree::Configuration.public_key = "your_public_key"
Braintree::Configuration.private_key = "your_private_key"

result = Braintree::Transaction.sale(
  :amount => "1000.00",
  :payment_method_nonce => nonce_from_the_client,
  :options => {
    :submit_for_settlement => true
  }
)
```

Therefore, I think I could make a modification to hack the `Affirm::Charge.create` method and create a default gateway for its client meanwhile keeping the `Affirm::ChargeGateway` for supporting multiple accounts. And it may be like this as follow.

For application which only has one affirm account

```ruby
Affirm::API.public_key = "xxx"
Affirm::API.secret_key = "xxx"
Affirm::API.api_url    = "https://sandbox.affirm.com/api/v2/"


charge = Affirm::Charge.create("checkout_token")
charge = Affirm::Charge.retrieve("ABCD-ABCD")
```

For application which might need to support multiple other services.

```ruby
charge_gateway = Affirm::ChargeGateway.new(
  environment: 'live', # or sandbox
  public_key: 'xxxx',
  secret_key: 'xxxx'
)

charge = charge_gateway.create("checkout_token")
charge = charge_gateway.retrieve("ABCD-ABCD")
```

How do you think about this design. Which one you think it might be better for needing to specify a custom gateway.

First

```ruby
charge_gateway = Affirm::ChargeGateway.new(
  environment: 'live', # or sandbox
  public_key: 'xxxx',
  secret_key: 'xxxx'
)

charge = charge_gateway.create("checkout_token")
charge = charge_gateway.retrieve("ABCD-ABCD")
```

Second

```
charge_gateway = Affirm::ChargeGateway.new(
  environment: 'live', # or sandbox
  public_key: 'xxxx',
  secret_key: 'xxxx'
)

charge = Affir::Charge.create("checkout_token", gateway: charge_gateway)
charge = Affir::Charge.retrieve("ABCD-ABCD", gateway: charge_gateway)
```

**Pull request closed without merging**